obsah/data_types.py

```diff
@@ -0,0 +1,69 @@
+"""
+obsah data types to be used to validate user input
+"""
+
+import argparse
+import inspect
+import os
+import sys
+
+
+class BaseType:
+    """
+    Base type class
+    """
+
+    @property
+    def name(self):
+        """
+        the name of the type
+        this is either the class name of, if set, the `type_name` property
+        """
+        if hasattr(self, 'type_name'):
+            return self.type_name
+        return self.__class__.__name__
+
+    def validate(self, string):
+        """
+        validate user input
+        """
+        return string
+
+class Boolean(BaseType):
+    """
+    Boolean type, accepting true/false and 1/0 only
+    """
+    def validate(self, string):
+        if string.lower() in ['true', '1']:
+            return True
+        if string.lower() in ['false', '0']:
+            return False
+        raise ValueError
+
+
+class Path(BaseType):
+    """
+    Path type, accepting anything that looks like a path
+    """
+    def validate(self, string):
+        return string
```

Comment:
Should this use [PurePath](https://docs.python.org/3/library/pathlib.html#pure-paths) to validate and normalize?

Reply:
yeah, possible.

don't look too closely at the individual implementations, there are mostly there to show "it works" ;)